### PR TITLE
Refactor and Enhance ListCommand Test Suite #154

### PR DESCRIPTION
**Pull request by sambhavnoobcoder**
## Description:

This pull request significantly improves the test coverage and accuracy of the ListCommand test suite. We've refactored existing tests and added new assertions to ensure that the ListCommand class behaves as expected under various scenarios.

## Changes Made:

1. test_list_assets:
   - Updated to verify correct invocation of list_assets method
   - Modified assertion to check for positional argument (max_assets) instead of keyword argument
   - Ensures max_assets parameter is correctly passed

2. test_list_models:
   - Implemented mock patching for ListCommand.list_models
   - Added assertion to verify list_models is called once without arguments
   - Ensures correct delegation when 'models' type is specified

3. test_list_apps:
   - Refactored to use mock patching and output capturing
   - Implemented mock side_effect to simulate expected output
   - Added assertions to verify both method call and correct output format

4. test_list_command_invalid_type:
   - Completely revamped to align with actual ListCommand behavior
   - Implemented comprehensive mocking for all list methods
   - Added checks to ensure no methods are called and no output is produced for invalid types

## Why These Changes Were Made:

1. To align tests with the actual implementation of ListCommand
2. To improve test coverage, especially for edge cases
3. To catch potential silent failures in the ListCommand class
4. To ensure robustness against unexpected inputs

## Test Results:

All tests pass successfully, demonstrating that:
- ListCommand correctly handles 'assets', 'models', and 'apps' types
- Proper arguments are passed to respective methods
- Expected output is produced for 'apps' listing
- Invalid types are handled gracefully (no action taken)

## Results :

The successful passing of the test suite will appear as follows :

Screenshot 2024-07-30 at 12 35 11 AM

Here , each of the four dots marks the successful passing of a test case , along with the time taken to run the tests and a final OK indicating successful passing of the test cases .

### CODE REVIEW

File: tests/list_command_test.py [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
+
+    def test_list_apps(self):
+        with patch('sys.stdout', new=io.StringIO()) as fake_out:
+            with patch.object(ListCommand, 'list_apps') as mock_list_apps:
+                mock_list_apps.side_effect = lambda: print("1: VS_CODE, 1.17.0, MACOS\n2: PIECES_FOR_DEVELOPERS, 3.0.2, MACOS")
+                ListCommand.list_command(type='apps')
+                
+            output = fake_out.getvalue().strip()
+            self.assertIn('1: VS_CODE, 1.17.0, MACOS', output)
+            self.assertIn('2: PIECES_FOR_DEVELOPERS, 3.0.2, MACOS', output)
```
Comment (Bishoy-at-pieces) [inline tests/list_command_test.py:30]: Can we make it a bit more dynamic here (use the platform library)

Comment (Bishoy-at-pieces) [reply tests/list_command_test.py:30]:
```python
 local_os = platform.system().upper() if platform.system().upper() in ["WINDOWS","LINUX","DARWIN"] else "WEB"
local_os = "MACOS" if local_os == "DARWIN" else local_os
```

**Review by Bishoy-at-pieces — CHANGES_REQUESTED** [review]

File: tests/list_command_test.py (hunk 1 above)
Comment (sambhavnoobcoder) [reply tests/list_command_test.py]: Addressed in [b4234d7](https://github.com/pieces-app/cli-agent/pull/159/commits/b4234d7e5dd16c355867b8883b1bb705eb4d8b39) commit

**Review by Bishoy-at-pieces — APPROVED** [review]
LGTM

**Review by mark-at-pieces — APPROVED** [review]
